src/web/client/utilities/fileAndEntityUtil.ts

```diff
@@ -31,13 +31,16 @@ export function getFileEntityName(fileFsPath: string) {
 
 export function getFileRootWebPageId(fileFsPath: string) {
     const entityId = getFileEntityId(fileFsPath);
-    return (
+
+    const rootWebPageId =
         (WebExtensionContext.entityDataMap.getEntityMap.get(entityId)
             ?.rootWebPageId as string) ??
         (WebExtensionContext.getVscodeWorkspaceState(fileFsPath)
             ?.rootWebPageId as string) ??
-        ""
-    );
+        "";
+
+    // If rootWebPageId is not present, return entityId as content is not localized
```

Comment:
what do you mean by this comment? Per this code - in case of webpage you have a rootPageId which will not be the case for other entities

Reply:
Webpages are localised, hence we have two different guid content page id root page id, so as other entities are not localised therefore storing same guid in root page id and entity page id.

Reply:
Be careful here - in case of content snippets as well content can be localized

Reply:
yes that's why it is like a fallback scenario, that if we don't receive any rootPageId then make it same as entity Id